ckan/lib/base.py

```diff
@@ -76,17 +76,31 @@ def abort(status_code=None, detail='', headers=None, comment=None):
                   comment=comment)
 
 
-def render_snippet(template_name, **kw):
+def render_snippet(*template_names, **kw):
     ''' Helper function for rendering snippets. Rendered html has
     comment tags added to show the template used. NOTE: unlike other
     render functions this takes a list of keywords instead of a dict for
-    the extra template variables. '''
+    the extra template variables.
+
+    :param template_names: the template to render, optionally with fallback
+        values, for when the template can't be found. For each, specify the
+        relative path to the template inside the registered tpl_dir.
+    :type template_names: str
+    :param kw: extra template variables to supply to the template
+    :type kw: named arguments of any type that are supported by the template
+    '''
 
-    output = render(template_name, extra_vars=kw)
-    if config.get('debug'):
-        output = ('\n<!-- Snippet %s start -->\n%s\n<!-- Snippet %s end -->\n'
-                  % (template_name, output, template_name))
-    return literal(output)
+    for template_name in template_names:
+        try:
+            output = render(template_name, extra_vars=kw)
+            if config.get('debug'):
+                output = ('\n<!-- Snippet %s start -->\n%s\n<!-- Snippet %s end -->\n'
+                        % (template_name, output, template_name))
+            return literal(output)
+        except TemplateNotFound:
```

Comment:
is there a way to make sure that the TemplateNotFound came from the first-referenced snippet and not some other template that snippet renders? If we catch both we could hide real template errors that should be raised

Reply:
Sure, I've added that in now.